zenodo/modules/records/serializers/schemas/common.py

```diff
@@ -305,6 +315,26 @@ def validate_language(self, value):
                 field_name=['language']
             )
 
+    @validates('dates')
+    def validate_dates(self, value):
+        """Validate that start date is before the corresponding end date."""
+        for interval in value:
+            start = arrow.get(interval.get('start')).date() \
```

Comment:
Arrow is a bit "liberal" with parsing stuff, so it would be better to be more explicit/strict about the calls:

```python
import arrow

arrow.get(None)
# <Arrow [2019-04-04T13:52:47.838153+00:00]>
arrow.get('2019')  # will do weird unix timestamp parsing...
# <Arrow [1970-01-01T00:33:39+00:00]>
arrow.get('2019-03')  # not necessarily bad...
# <Arrow [2019-03-01T00:00:00+00:00]>

# Strict format:
arrow.get(None, 'YYYY-MM-DD')
# TypeError: Can't parse two arguments of types '<type 'NoneType'>', '<type 'str'>' 
arrow.get('2019', 'YYYY-MM-DD')
# ParserError: Failed to match '(?P<YYYY>\d{4})\-(?P<MM>\d{2})\-(?P<DD>\d{2})' when parsing '2019'
arrow.get('2019-03', 'YYYY-MM-DD')  # we lost this...
# ParserError: Failed to match '(?P<YYYY>\d{4})\-(?P<MM>\d{2})\-(?P<DD>\d{2})' when parsing '2019-03'
arrow.get('2019-03-17', 'YYYY-MM-DD')  # we won this...
# <Arrow [2019-03-17T00:00:00+00:00]>
```

Check the [arrow docs](https://arrow.readthedocs.io/en/latest/) and play around with the format parsing to make sure we don't do anything weird... 

If you find other places in the codebase (even if they're not relevant to this PR), create an issue so we can have a look later.